api/src/main/resources/client_api.yaml

```diff
@@ -113,7 +113,12 @@ definitions:
           Maximum number of results to return in a response. Defaults to 1000.
         type: integer
         format: int32
-      # TODO: add fieldSpec, output information
+      fieldSet:
+        description: >
+          Specification defining what data to return for participants in the cohort. Defaults to
```

Comment:
Do all tables in the CDR have rows corresponding to a participant? Something like the concepts table I imagine wouldn't. How does that interaction work and what's the rule here? Does this just apply to talbes that have a direct foreign key to participants?

Reply:
So a cohort is a set of participants; they have IDs (found in a bunch of tables including the person table.)

All the OMOP CDM tables do have a person ID. There are other tables (e.g. concept) which don't have person IDs... but it wouldn't make sense to use this API for accessing them, since the point of it is to materialize a cohort -- pick out a subset of rows based on participant ID.

If you don't specify a field set at all, you will just get back participant IDs, which is what this comment is about.

Reply:
(We could instead say that moving forward a FieldSet is required. I don't really mind on that one.)

Reply:
Optional fieldset seems OK to me; if for no other reason than it allows you to test a very simple query to kick the tires.

> All the OMOP CDM tables do have a person ID

Is this true? I only have a surface-level understanding of OMOP so I'm not sure on terminology here. I did find this DDL which contains tables which don't have person IDs. https://github.com/OHDSI/CommonDataModel/blob/master/BigQuery/OMOP%20CDM%20bigquery%20ddl.txt

Could you add more documentation here or below to FieldSet? What you're saying makes sense, but it seems like it would be a useful thing to make explicit. Unless the link to the CDM schema you put below already restricts you to the right subset of tables with person_ids.

Reply:
You're right, I misspoke. There are tables in the CDM that don't have person IDs. Updated the comments to indicate its only the ones that do we support.